lib/matplotlib/colorbar.py

```diff
@@ -674,9 +674,9 @@ def _process_values(self, b=None):
 
             b = self.norm.inverse(self._uniform_y(self.cmap.N + 1))
             if self._extend_lower():
-                b[0] = b[0] - 1
+                b[0] = 0.9 * b[0]
```

Comment:
Maybe the nicest way to handle this would be to have the norm itself define two special values, one such that its norm is less than zero, and the other such that its norm is greater than one.  Then the colorbar code here could just plug in those values as needed.